apis/nucleus/src/components/listbox/ListBoxPortal.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import ListBoxInline from './ListBoxInline';
+
+export default function ListBoxPortal({ app, fieldIdentifier, stateName, element, options }) {
+  let returnComponent = null;
+  const isFrequencyMaxNeeded = options.histogram || options.frequencyMode !== 'N';
+  if (fieldIdentifier.qLibraryId && isFrequencyMaxNeeded) {
+    returnComponent = (
+      <ListBoxFetchMasterItem
+        app={app}
+        fieldIdentifier={fieldIdentifier}
+        stateName={stateName}
+        element={element}
+        options={options}
+      />
+    );
+  } else {
+    returnComponent = (
+      <ListBoxInline
+        app={app}
+        fieldIdentifier={fieldIdentifier}
+        stateName={stateName}
+        element={element}
+        options={options}
+      />
+    );
```

Comment:
If the components will get identical arguments we *could* also do:
```
const TheComponent = fieldIdentifier.qLibraryId && isFrequencyMaxNeeded ? ListBoxFetchMasterItem : ListBoxInline;
return <TheComponent
        app={app}
        fieldIdentifier={fieldIdentifier}
        stateName={stateName}
        element={element}
        options={options}
      />

```
(depends on if they will diverge later on I guess…)

Reply:
Nice! :)